process.c

```diff
@@ -4679,56 +4679,132 @@ rb_spawn(int argc, const VALUE *argv)
+ *  - +true+ if the command exits with status zero.
+ *  - +false+ if the exit status is a non-zero integer.
+ *  - +nil+ if the command could not execute.
  *
- *  If the <code>exception: true</code> argument is passed, the method
- *  raises an exception instead of returning +false+ or +nil+.
+ *  Raises an exception (instead of returning +false+ or +nil+)
+ *  if keyword argument +exception+ is set to +true+.
  *
- *  The arguments are processed in the same way as
- *  for Kernel#spawn.
+ *  Assigns the command's error status to <tt>$?</tt>.
  *
- *  The hash arguments, env and options, are same as #exec and #spawn.
- *  See Kernel#spawn for details.
+ *  The new process is created using the
+ *  {system system call}[https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/functions/system.html];
+ *  it may inherit some of its environment from the calling program
+ *  (possibly including open file descriptors).
  *
- *     system("echo *")
- *     system("echo", "*")
+ *  Argument +env+, if given, is a hash that affects +ENV+ for the new process;
+ *  see {Execution Environment}[rdoc-ref:Process@Execution+Environment].
  *
- *  <em>produces:</em>
+ *  Argument +options+ is a hash of options for the new process;
+ *  see {Execution Options}[rdoc-ref:Process@Execution+Options].
+ *
+ *  The first required argument is one of the following:
+ *
+ *  - +command_line+ if it is a string,
+ *    and if it begins with a shell reserved word or special built-in,
+ *    or if it contains one or more metacharacters.
+ *  - +exe_path+ otherwise.
+ *
+ *  <b>Argument +command_line+</b>
+ *
+ *  \String argument +command_line+ is a command line to be passed to a shell;
+ *  it must begin with a shell reserved word, begin with a special built-in,
+ *  or contain meta characters:
+ *
+ *    system('echo')                                  # => true  # Built-in.
+ *    system('if true; then echo "Foo"; fi')          # => true  # Shell reserved word.
+ *    system('date > /tmp/date.tmp')                  # => true  # Contains meta character.
+ *    system('date > /nop/date.tmp')                  # => false
+ *    system('date > /nop/date.tmp', exception: true) # Raises RuntimeError.
+ *
+ *  Assigns the command's error status to <tt>$?</tt>:
+ *
+ *    system('echo')                             # => true  # Built-in.
+ *    $?                                         # => #<Process::Status: pid 640610 exit 0>
+ *    system('date > /nop/date.tmp')             # => false
+ *    $?                                         # => #<Process::Status: pid 640742 exit 2>
+ *
+ *  The command line may also contain arguments and options for the command:
+ *
+ *    system('echo "Foo"') # => true
+ *
+ *  Output:
+ *
+ *    Foo
+ *
+ *  On a Unix-like system, the shell is <tt>/bin/sh</tt>;
+ *  otherwise the shell is determined by environment variable
+ *  <tt>ENV['RUBYSHELL']</tt>, if defined, or <tt>ENV['COMSPEC']</tt> otherwise.
```

Comment:
Nit: I'm not sure if "otherwise" is appropriate since `RUBYSHELL` is used only on Windows now (OS/2 is mentioned in ruby.1 but not supported).